zvedamrance/src/main/java/ServletAdd.java

```diff
@@ -0,0 +1,27 @@
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+@WebServlet("/user/add")
+public class ServletAdd extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        RequestDispatcher dispatcher = request.getRequestDispatcher("/adder.jsp");
+        dispatcher.forward(request, response);
+    }
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        User user = new User();
+        user.setUserName(req.getParameter("userName"));
+        user.setEmail(req.getParameter("userEmail"));
+        user.setPassword(req.getParameter("userPassword"));
+        UserDao userDao = new UserDao();
```

Comment:
CLEAN-CODE: pouzivat nemenne promenne vsude kde to je mozne v aplikaci.

```suggestion
final UserDao userDao = new UserDao();
```

**Duvod**

Používání neměnných proměnných final v Javě je dobré, protože zajišťuje, že hodnota proměnné se po jejím přiřazení nemůže změnit, což zvyšuje čitelnost a spolehlivost kódu. Pomáhá to také při optimalizaci kódu, protože kompilátor může lépe předvídat chování programu. Dále zajišťuje, že nebudou neúmyslné změny hodnoty proměnných, což snižuje riziko chyb, a také to usnadňuje práci s multithreadingem, kde je důležité, aby hodnoty nebyly modifikovány souběžně v různých vláknech.